plugins/module_utils/query.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```

Comment:
`module_utils` code should have BSD license (reason: historic).

Reply:
I know module utils from ansible/ansible are licensed under BSD in order to allow modules to be non-GPL. But this logic does not hold for the ITSM collection module utils since we know our modules are GPLv3. It is also worth noting that all other module utils are already under GPLv3 so making this code licensed under BSD would introduce some inconsistencies into the collection itself.

DO you still want to have this file under BSD?

Reply:
@abadger Do you know if we can make all `module_utils` code as GPLv3 as per the above reason? or is there is a hard requirement of making `module_utils` as BSD?

Reply:
@tadeboro Please don't merge this PR until we resolve this.

Reply:
https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing is the document that describes the license requirements. Citing relevant part of that document:

> `module_utils` must be licensed with a free software license that is compatible with the GPL-3.0-or-later. Ansible itself typically uses the BSD-2-clause license to make it possible for third-party modules which are licensed incompatibly with the GPLv3 to use them. Please consider this use case when licensing your own module_utils.

So GPLv3 is fine here. We could license our code under a more permissive license if we would like to allow people to use our code in code that is not GPLv3, but since the code here is really ITSM-collection-specific (there already exists a Python client in the wild, so there is no need to use our custom client), I see no real benefit in relicensing the code.

Reply:
Got it. Thanks for the information.